spec/acceptance/api/line_item_plan_cost_spec.rb

```diff
@@ -1,9 +1,9 @@
 require 'rails_helper'
 
 resource "LineItem", transactions: false do
-  let(:provider) { FactoryBot.create(:simple_provider) }
-  let(:buyer) { FactoryBot.create(:simple_buyer, provider_account: provider) }
-  let(:application) { FactoryBot.create(:simple_cinstance, user_account: buyer) }
+  let(:provider) { FactoryBot.create(:provider_account) }
+  let(:buyer) { FactoryBot.create(:buyer_account, provider_account: provider) }
+  let(:application) { FactoryBot.create(:cinstance, user_account: buyer) }
```

Comment:
They don't make sure objects have proper hierarchy. Specifically `:simple_cinstance` will create a plan that has another provider account than the `user_account` specified, so it will have `tenant_id` mismatch.

I don't understand why we need the `simple_*` factories. I assume when we don't need full correctness. So I see no point to replicate the proper logic in the `simple_*` factories. If you need full correctness, use the normal factories.

To me these `simple_*` factories are confusing. Maybe we should remove them. idk